### ISSUE TITLE
Excepting ids and passwords from the Redundant Entry SC

### ISSUE THREAD

**Issue by guyhickling** [w3c/wcag#1080]
It is a common practice when entering data during important online financial and other transactions, performed after the user has logged in to a system, for the system to require them to log in again before submitting their input data. This is done to avoid the scenario where a user logs in, then leaves their computer unattended, at which point someone else could come along and enter a transaction, or change the user's profile data or password, whether for fraudulent purposes or simply as a joke.

Clearly the autocompletion of the password field, as expected by SC1.3.5, cannot be allowed in this kind of re-authenification, for security reasons. It would defeat the object of logging in again. The user must type it in afresh and many designers (depending on the kind of system being designed and how important the transaction is), would want the same for the id field(s) as well. 

And just as a point of clarity and grammatical correctness, I think "required on subsequent" should be "in" not "on". And a couple of commas round the "entered by or" clause would improve clarity as well.

So I suggest adding the following additional exception and changes to the new Redundant Entry SC:

> add ".....or when required for authentification".

> change the word "Exception" to "Exceptions"

> and amend the wording slightly to "...For steps in a process, information, previously entered by or provided to the user, that is required in subsequent...."

All of which brings us to SC1.3.5; that should have a similar exception for ids and passwords only we didn't think of it at the time.

**Comment by patrickhlauke** [comment]
for 1.3.5, i'd point out that autocompletion is a side effect of the chosen use of `autocomplete` attribute. it is not the requirement of the SC per se, which asks that fields be programmatically identified (and theoretically, that could be accomplished with something other than an `autocomplete` attribute, though that is...a chicken/egg theoretical exercise at the moment). so i don't think the exception should/could be added to 1.3.5 (though from a realistic development point of view, i can understand completely that devs may willingly fail 1.3.5 in favour of security)

**Comment by guyhickling** [comment]
Patrick, I think your words "...devs may willingly fail 1.3.5 in favour of security" say it all for that criterion. But the above exception is needed in the new Redundant Entry SC so perhaps we can concentrate on that. We don't want any more SCs that devs "willingly fail" for security reasons!

**Comment by alastc** [comment]
> Clearly the autocompletion of the password field, as expected by SC1.3.5, cannot be allowed in this kind of re-authenification, for security reasons.

I don't think that is correct, for 1.3.5 we did [consider username/password security](https://github.com/w3c/wcag/issues/407#issuecomment-401418096) (and get a security review). If you have a password manager / browser filling in your username password there is no reason to block that for initial entry, therefore not for re-entry either. (See also the NCSC guidance: [Let them paste passwords](https://www.ncsc.gov.uk/blog-post/let-them-paste-passwords).)

We also went through those conversations again for Accessible Authentication, having autocomplete is a useful technique there, but there are other options as well.

For Redundant Entry, I've been thinking about re-authentication as a mechanism to get you back to a process, rather than a step in the process itself. I don't think it particularly matters though as the *site* re-filling the authentication obviously undermines the point of authenticating, so fits under 'essential'. We can include that example in the understanding doc if that helps?

**Comment by guyhickling** [comment]
> We can include that example in the understanding doc if that helps?

Yes I guess, as you say, it comes under the heading of essential. So if this kind of re-authentication is mentioned in the Understanding as an example of "essential", that would meet the need so far as I'm concerned. It is something I see quite often in banking and other financial websites, so there is good reason for mentioning it. Will this happen resulting from this thread or does it need to be initiated somewhere else?

**Comment by alastc** [comment]
This is the last paragraph of the [understanding doc](https://raw.githack.com/w3c/wcag/wcag22-redundant-entry/understanding/22/redundant-entry.html):
> There is an exception for essential uses of input re-entry for things like password entry (security), or memory games which would not be possible if the previous answers were supplied.

**Comment by guyhickling** [comment]
Yes, that covers it, thank you. This issue can be closed.

**Comment by jasonday** [comment]
> This is the last paragraph of the [understanding doc](https://raw.githack.com/w3c/wcag/wcag22-redundant-entry/understanding/22/redundant-entry.html):
> 
> > There is an exception for essential uses of input re-entry for things like password entry (security), or memory games which would not be possible if the previous answers were supplied.

This has been removed from the latest understanding doc - https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose.html

**Comment by patrickhlauke** [comment]
this thread is about redundant entry, not identify input purpose, @jasonday
